### PR TITLE
[Merged by Bors] - feat: add missing Set.Ioo_inter_Iio etc

### PR DESCRIPTION
**Pull request by sven-manthe**
add missing lemmas. I'm not sure if these should be @[simp], as e.g. Ico_inter_Ico isn't, but Ico_inter_Iio is

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — COMMENTED** [review]
Can you write the `ext` on a separate line? Apart from that, looks good to me. (I'm not sure about `simp` either.)

**Review by Vierkantor — COMMENTED** [review]
Some of the lemmas in this file use `sup` and `inf`, some use `min` and `max`. We should be consistent. I *think* `inf` and `sup` are better (see e.g. this [Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/min.2Fmax.20definitions.20non-alignment)), but unsure.

@urkud @RemyDegenne, you authored many of the lemmas in this file, do you have opinions?

**Review by fpvandoorn — APPROVED** [review]
I think we can merge this. We can unify the min/max/sup/inf situation in a later PR.

**Comment by Vierkantor** [comment]
Fair enough, the confusion was already there so it shouldn't be the responsibility of this PR to fix it. I've started [a Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/.60sup.60.2F.60inf.60.20or.20.60max.60.2F.60min.60.20for.20set.20interval.20lemmas.3F) so we don't forget about cleaning this up.

Since the other results adjacent to this don't have `simp`, these probably shouldn't either.

In conclusion, looks good to me, let's get this merged!

bors r+
